zkapp/contract.ts

```diff
@@ -51,21 +57,29 @@ export class HelloWorld extends SmartContract {
     this.emitEvent('singleField', x);
   }
 
-  @method async emitStructEvent() {
-    const x = this.x.getAndRequireEquals();
-    const y = this.y.getAndRequireEquals();
-    const z = this.z.getAndRequireEquals();
-    this.emitEvent(
-      'struct',
-      new TestStruct({ x, y, z, address: this.address })
-    );
+  @method async emitStructEvent(struct: TestStruct) {
+    this.emitEvent('struct', struct);
+  }
+
+  @method async emitStructsEvent(structs: TestStructArray) {
```

Comment:
Updating these methods give us more variance to test with - not just emitting the same thing over and over.